src/components/Checkbox.tsx

```diff
@@ -0,0 +1,123 @@
+import type { CheckboxProps, ToggleProps } from '@react-types/checkbox';
+import { gsap } from 'gsap';
+import type { ChangeEvent, ComponentPropsWithRef } from 'react';
+import { useEffect, useRef } from 'react';
+import { useCheckbox } from 'react-aria';
+import { useToggleState } from 'react-stately';
+import { styled } from '../../stitches.config';
+import { CheckIcon } from '../animationIcons/CheckIcon';
+
+const Label = styled('label', {
+  '& input:checked:disabled + div': {
+    color: '$gray50',
+  },
+  display: 'block',
+  position: 'relative',
+});
+
+const Input = styled('input', {
+  '&:checked': {
+    backgroundColor: '$brandYellow',
+    borderColor: '$brandYellow',
+  },
+  '&:checked&:disabled': {
+    backgroundColor: '$yellow20',
+    borderColor: '$yellow20',
+  },
+  '&:disabled': {
+    borderColor: '$gray30',
+  },
+  '&:focus': {
+    outline: '2px solid $brandYellow',
+    outlineOffset: '2px',
+  },
+  '&:indeterminate': {
+    borderColor: '$brandYellow',
+    position: 'relative',
+  },
+  '&:indeterminate:after': {
+    backgroundColor: '$brandYellow',
+    content: '""',
+    height: '12px',
+    left: '50%',
+    position: 'absolute',
+    top: '50%',
+    transform: 'translate(-50%, -50%)',
+    width: '12px',
+  },
+  '&:invalid': {
+    borderColor: '$uiErrorRegular',
+  },
+  appearance: 'none',
+  backgroundColor: '$brandWhite',
+  borderColor: '$gray50',
+  borderRadius: '$4',
+  borderStyle: '$solid',
+  borderWidth: '$thin',
+  height: '24px',
+  margin: '0',
+  position: 'relative',
+  width: '24px',
+});
+
+const IconContainer = styled('div', {
+  left: '50%',
+  position: 'absolute',
+  top: '50%',
+  transform: 'translate(-50%, -50%)',
+});
+
+type CustomCheckboxProps = {
+  text?: string;
+};
+
+export const Checkbox = ({
+  validationState,
+  text,
+  defaultSelected,
+  isIndeterminate,
+  ...restProps
+}: CheckboxProps &
+  ComponentPropsWithRef<'input'> &
+  CustomCheckboxProps &
+  ToggleProps) => {
+  const ref = useRef<HTMLInputElement>(null);
+  const state = useToggleState({ defaultSelected, validationState });
```

Comment:
A small note on convention. The `useToggleState` hook is not actually giving you back the state of the toggle; rather, it is returning an object with a `setSelected` function (to mutate the state) and a `toggled` value that is actually the state of the toggle. Let's rewrite this to say `const { toggled, setSelected } = ...` and use destructuring to make this a bit clearer.

(This is a bit closer to react convention, even if it sounds like a nitpick 🙏 )

Reply:
@Boeing787 Yep I can see that this makes it clearer what's going on here - I've now refactored to destructure the `ToggleProps` from the `useToggleStateHook`